src/sentry/integrations/notifydisable.py

```diff
@@ -0,0 +1,56 @@
+from sentry.models import Organization
+from sentry.services.hybrid_cloud.integration import RpcIntegration
+
+provider_types = {
+    "integration": "integrations",
+    "plugin": "plugins",
+    "sentry-app": "sentry-apps",
+}
+
+
+def get_url(organization: Organization, provider_type: str, provider) -> str:
+    type_name = provider_types.get(provider_type, "")
+    return str(
+        organization.absolute_url(
+            f"/settings/{type_name}/{provider}/",
+        )
+    )
+
+
+def get_provider_type(redis_key) -> str:
+    for provider in provider_types:
+        if provider in redis_key:
+            return provider
```

Comment:
What if it's not found?

Reply:
Oh and since you're adding typing, you can add `redis_key: str` to this

Reply:
Since we call `notify_disable` only when we have a valid `redis_key` and all redis_keys have one of the providers in them we shouldn't have that case. It could return an empty str but that will make a bad url so maybe we should throw and error?

Reply:
Maybe worst case we just remove the provider from the URL so it goes to the integrations directory, though that shouldn't happen. We just need to return something here in case the if statement fails.

Reply:
This function determines which directory to go to (sentry app, plugin or integration). I changed it to return `integration` which might create a bad url if the provider isn't an integration, like https://sentry.sentry.io/settings/integrations/twillio
If we were to return no string like https://sentry.sentry.io/settings/twillio it would rereroute to https://sentry.sentry.io/settings/organization/